packages/wonder-stuff-server/src/types.js

```diff
@@ -0,0 +1,122 @@
+// @flow
+import type {
+    NpmLogLevels,
+    Logger as WinstonLogger,
+    Info as WinstonInfo,
+    Transport,
+} from "winston";
+
+import type {$Request} from "express";
+
+/**
+ * Describes logging metdata.
+ */
+export type Info = WinstonInfo<NpmLogLevels>;
+
+/**
+ * Defines the different log levels.
+ */
+export type LogLevel = $Keys<NpmLogLevels>;
+
+/**
+ * Describes the interface for logging gateway activity.
+ */
+export type Logger = WinstonLogger<NpmLogLevels>;
+
+/**
+ * The runtime modes that a gateway can run under.
+ */
+export enum Runtime {
+    // TODO(somewhatabstract, FEI-4172): Update eslint-plugin-flowtype when
+    // they've fixed https://github.com/gajus/eslint-plugin-flowtype/issues/502
+    /* eslint-disable no-undef */
+    Production = "production",
+    Development = "development",
+    Test = "test",
+    /* eslint-enable no-undef */
+}
+
+/**
+ * Options to configure logging.
+ */
+export type LoggingOptions = {
```

Comment:
New: replaces the args to the createLogger method (which were just mode and level).